### DISCUSSION TITLE
Bumping WiX Dev .NET version to latest LTS on each Major Release?

### DISCUSSION THREAD

**Discussion by bevanweiss** [wixtoolset/issues#8659] [WiX Development]
It would be great if it were possible to use some of the newer syntactic sugar available from the newer .NET releases (like [collection expressions](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/operators/collection-expressions)).

Is there significant effort with updating the various project versions (and dependencies) with each major release?

I would think that the various external consumption libraries (like DTF, etc) would still be targeted to either .NET Standard 1.1 or .2.0, but things like the integration tests, compiler/decompiler etc could be .NET LTS (i.e. currently .NET v8, and .NET v10 in a year or so).

**Comment by barnson** [comment]
Build tools still need to run on .NET Framework.

**Reply by bevanweiss** [reply]
I was initially just thinking those projects that are currently at .net6.0 could be bumped up to .net8.0
I'm not aware of any loss of backwards compatibility from 6->8, both are built on .NET Standard 2.1, and I'm pretty sure both can target back to .NET Standard 1.1

I haven't dealt with .EXE generation on .net5+ to target .net FW however, only libraries.
I'll give it a try.

**Reply by barnson** [reply]
Sorry, I was thinking of language version; that's locked because later language features aren't supported on .NET Framework. We'll likely retarget when versions of .NET go out of service.

**Reply by bevanweiss** [reply]
I think even on the Language Version front there's quite a few oddities.  Lots of the newer features (like Collection Initializers) are changes to the C#/F# 'language's, but not to the Runtime, and so are backwards compatible.  Because once compiled down to IL, they are still valid in the legacy runtimes.

.NET 6.0 goes out of support in November
https://dotnet.microsoft.com/en-us/platform/support/policy

**Comment by robmen** [comment]
Collection expressions are not a big enough feature to force the move from .NET 6 to .NET 8. The timing is such that .NET 6 is good for WiX v6, and WiX v7 will probably move to .NET 8 unless there is something compelling in .NET 10.

**Reply by barnson** [reply]
.NET 6 is EOL in November this year.

**Reply by bevanweiss** [reply]
It'd be great if either of you could give my [PR](https://github.com/wixtoolset/wix/pull/554) around this a try. 
It's potentially updating some elements that you might not want updated, but it should 'just work'..

**Reply by barnson** [reply]
I'm waiting for Rob's dissenting opinion.

**Reply by bevanweiss** [reply]
I’m hoping his reluctance towards .NET 8 came from pain around MSBuild stuff… which should be all sorted with this PR (and the System.Memory alignment).

**Reply by robmen** [reply]
No. As I noted, I'm not interested in forcing customers forward to .NET 8 in WiX v6.

**Reply by robmen** [reply]
By the way, I do appreciate you running down the MSBuild nonsense. Just know that it is our goal to get rid of heat, not move it forward.  Getting the `Files` element into the world was a very big step in that direction.
